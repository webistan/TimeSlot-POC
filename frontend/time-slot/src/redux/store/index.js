import { applyMiddleware, createStore } from 'redux'

import { createLogger } from 'redux-logger'
import createSagaMiddleware from 'redux-saga'
import rootReducer from '../../redux/reducer'
import sagas from '../../redux/saga'

// import sagas from './redux/saga';
// import rootReducer from './redux/reducer';

const loggerMiddleware = createLogger()

const sagaMiddleware = createSagaMiddleware()

const store = createStore(rootReducer, applyMiddleware(sagaMiddleware,loggerMiddleware))

sagaMiddleware.run(sagas)

export default store

