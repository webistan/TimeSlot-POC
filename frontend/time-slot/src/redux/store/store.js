import { configureStore } from "@reduxjs/toolkit";
import slot from "../reducer/index";

export const createStore = () =>
  configureStore({
    reducer: {
      slotReducer: slot,
    },
  });
