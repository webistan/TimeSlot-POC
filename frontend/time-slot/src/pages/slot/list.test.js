// Importing the jest testing library
import '@testing-library/jest-dom'

import { cleanup, fireEvent, render, screen } from "@testing-library/react";

import { Provider } from 'react-redux';
import SlotList from "./List";
import {createStore} from "../../redux/store/store";
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';

// afterEach function runs after each test suite is executed
afterEach(() => {
    cleanup(); // Resets the DOM after each test suite
})
  
describe("SlotList Component" , () => {
    const data = {start_date: '2022-11-18', end_date: '2022-11-30'}

    test("Should show empty inputs when no values are provided", () => {
      const { getByTestId } = render(
        <Provider store={createStore()}>
          <SlotList />
        </Provider>
      );
      const startDateInput = getByTestId('startDate');
      const endDateInput = getByTestId('endDate');
  
      expect(startDateInput).toBeInTheDocument(); 
      expect(endDateInput).toBeInTheDocument(); 
    });

    test('Should show filled in inputs when values are provided', () => {

      const { getByTestId } = render(
        <Provider store={createStore()}>
          <SlotList />
        </Provider>
      );
      const startDateInput = getByTestId('startDate');
      const endDateInput = getByTestId('endDate');
      
      userEvent.type(startDateInput, '2022-11-18');
      userEvent.type(endDateInput, '2022-11-30');
      expect(startDateInput).toHaveValue('2022-11-18');
      expect(endDateInput).toHaveValue('2022-11-30');

      const SlotListDomTree = renderer.create(<Provider store={createStore()}>
        <SlotList />
      </Provider>).toJSON();
      expect(SlotListDomTree).toMatchSnapshot();

    });


    // test("Should not allow end date input lower than start date", () => {

    //   const { getByTestId } = render(
    //     <Provider store={createStore()}>
    //       <SlotList />
    //     </Provider>
    //   );
    //   const startDateInput = getByTestId('startDate');
    //   const endDateInput = getByTestId('endDate');
      
    //   userEvent.type(startDateInput, '2022-11-18');
    //   userEvent.type(endDateInput, '2022-11-11');
    //   expect(startDateInput).toHaveValue('2022-11-18');
    //   expect(endDateInput).toHaveValue('2022-11-11');
    //   expect(endDateInput).toBeInvalid();
    // });
  


})
