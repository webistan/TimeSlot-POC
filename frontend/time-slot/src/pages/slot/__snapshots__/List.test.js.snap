// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlotList Component Should show filled in inputs when values are provided 1`] = `
<div
  className="container"
>
  <div
    className="wt-filter"
  >
    <h2>
      Manage Schedule
    </h2>
    <div
      className="filter-block"
      data-testid="date-picker"
    >
      <input
        className="filter"
        data-testid="startDate"
        id="startDate"
        name="startDate"
        onChange={[Function]}
        type="date"
      />
      <input
        className="filter"
        data-testid="endDate"
        id="endDate"
        name="endDate"
        onChange={[Function]}
        type="date"
      />
    </div>
  </div>
  <div
    className="wt-timeslots-list"
  >
    <h3>
      Timeslot
    </h3>
  </div>
  <div
    style={
      Object {
        "textAlign": "right",
      }
    }
  >
    <p
      className="errorClass"
    />
    <button
      className="button2 btn-lg"
      onClick={[Function]}
    >
      Save Schedule
    </button>
  </div>
</div>
`;
